src/intervals/intervals.jl

```diff
@@ -25,7 +25,8 @@ struct Interval{T<:Real} <: AbstractInterval{T}
                 new(a, b)
 
             else
-                throw(ArgumentError("Interval of form [$a, $b] not allowed. Must have a ≤ b to construct interval(a, b)."))
+                @warn "Invalid input, empty interval is returned"
```

Comment:
Would it be reasonnable to never check the validity here, get rid of the `validity_check` setting and document that to get IEEE compliant and thouroughly validated interval once should use either `interval` or `DecoratedInterval`?

Related but not directly in the scope of this PR, I wonder if it wouldn't be clearer for users to have `Interval` as the default IEEE compliant one and use some trick to get an `unsafe_interval` function for internal operations.

If you thing it is too tangential to the issue tackled here, I can add a separate issue.

Reply:
`validity_check` is indeed set by default to `false` and since you have functions like `..` and `interval` I think it could be removed without too much harm.

The `Interval` vs `interval` might be an interesting discussion. One thing that puzzles me is the asymmetry of bare and decorated intervals, in the sense that for bare the constructors `Interval` doesn't do any checks and you should use `interval` or `..`, but for decorated intervals the constructors `DecoratedIntervals` does checks (well a few issues there, next PR on the way for that :D ). I think it might be important to have a mechanism to construct "unsafe intervals" which merely puts the given inputs as lower and upper bound, at least for testing purposes. Maybe this can be addressed in a separate issue?